### PR TITLE
Algolia search button, placeholder, and manual live search switch

### LINKED ISSUE #316: Add search button and search bar placeholder in Algolia Searcher
Add search button and search bar placeholder

help kb search

Mockup copied from https://github.com/Yoast/yoast-components/issues/198

### PR DESCRIPTION
**Pull request by afercia**
## Summary

This PR can be summarized in the following changelog entry:

* Update the Algolia Search design and add ability to set it to manual or live search.

## Relevant technical choices:

* added `enableLiveSearch` prop to switch search mode, false by default
* the Search submit button is displayed only in the manual mode

## Test instructions

This PR can be tested by following these steps:

* use the following code in ContentAnalysis
* test in desktop and responsive view
* test in both manual and live mode, changing the `enableLiveSearch` prop value

```
import React from "react";
import styled from "styled-components";

import AlgoliaSearcher from "../../../AlgoliaSearch/AlgoliaSearcher";

export const ContentAnalysisContainer = styled.div`
	min-height: 700px;
	width: 100%;
	padding-top: 40px;
	background-color: white;
`;

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
export default function ContentAnalysis() {
	return <ContentAnalysisContainer>
		<AlgoliaSearcher maxWidth="1280px" enableLiveSearch={ false } />
	</ContentAnalysisContainer>;
}
```

Screenshots:

Desktop view:

screen shot 2017-09-29 at 11 24 00

Responsive view:

screen shot 2017-09-29 at 11 23 22

Fixes #316

### CODE REVIEW

File: composites/AlgoliaSearch/SearchBar.js [hunk 1]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
 const SearchBarWrapper = styled.div`
-	overflow: hidden;
-	width: 100%;
 	padding: 0 16px;
-	box-sizing: border-box;
+
+	form {
+		display: flex;
+
+		@media screen and (max-width: ${ breakpoints.mobile } ) {
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: space before `max-width`

File: composites/AlgoliaSearch/SearchBar.js [hunk 2]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+			flex-wrap: wrap;
+		}
+	}
+
+	@media screen and (max-width: ${ breakpoints.mobile } ) {
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: space before `max-width`

File: composites/AlgoliaSearch/SearchBar.js [hunk 3]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+			margin-top: 1em;
+		}
+	}
+`;
+
+const SearchHeading = styled.h2`
+	font-size: 1em;
+	margin: 0.5em 0 0.5em 58px;
+
+	@media screen and (max-width: ${ breakpoints.mobile } ) {
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: space before `max-width`

File: composites/AlgoliaSearch/SearchBar.js [hunk 4]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+		button {
+			min-width: 100%;
+			margin-top: 1em;
+		}
+	}
+`;
+
+const SearchHeading = styled.h2`
+	font-size: 1em;
+	margin: 0.5em 0 0.5em 58px;
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: Not on 8px grid. Does 56px good as well?

File: composites/AlgoliaSearch/SearchBar.js [hunk 5]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+
+	@media screen and (max-width: ${ breakpoints.mobile } ) {
+		margin-left: 0;
+	}
+`;
+
+const SearchLabel = styled.label`
+	flex: 0 0 42px;
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: Not on 8px grid. Does 40px look good as well?

File: composites/AlgoliaSearch/SearchBar.js [hunk 6]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+	height: 3em;
+	// it's already a flex item, let's make it also a flex container to align the svg icon
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: // Align the svg icon by making the label a flex container.

File: composites/AlgoliaSearch/SearchBar.js [hunk 7]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
+	align-items: center;
 `;
 
 const SearchBarInput = styled.input`
+	flex: 1 1 auto;
 	box-sizing: border-box;
-	height: 2.5em;
-	width: calc(100% - 2em);
+	height: 3em;
 	box-shadow: inset 0 2px 8px 0px rgba(0,0,0,0.3);
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: Not your code, but if possible, add spaces

File: composites/AlgoliaSearch/SearchBar.js [hunk 8]
```diff
@@ -7,43 +7,74 @@ import debounce from "lodash/debounce";
...
-	float: left;
-	padding-left: 16px;
-`;
+	margin-right: 24px;
+	padding: 0 8px 0 16px;
 
-const SearchHeading = styled.h2`
-	font-size: 1em;
-	margin: 0.5em 0;
+	@media screen and (max-width: ${ breakpoints.mobile } ) {
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: Space before `max-width`

File: composites/AlgoliaSearch/SearchBar.js [hunk 9]
```diff
@@ -137,7 +183,12 @@ class SearchBar extends React.Component {
 						autoCorrect="off"
 						autoCapitalize="off"
 						spellCheck="false"
+						placeholder={ placeholderText }
```
Comment (IreneStr) [inline composites/AlgoliaSearch/SearchBar.js]: Do we want the possibility to pass the placeholder text as a prop?

File: composites/AlgoliaSearch/tests/__snapshots__/AlgoliaSearcherTest.js.snap [hunk 10]
```diff
@@ -36,14 +36,24 @@ exports[`the AlgoliaSearcher component with headingText matches the snapshot 1`]
...
         id="kb-search-input"
         name="search-input"
         onChange={[Function]}
+        placeholder="Search the knowledge base"
         spellCheck="false"
         type="text"
       />
+      <button
+        className="sc-bwzfXH eeZBlJ sc-bdVaJa bsIiYo"
+        onClick={undefined}
```
Comment (IreneStr) [inline composites/AlgoliaSearch/tests/__snapshots__/AlgoliaSearcherTest.js.snap]: undefined

File: composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap [hunk 11]
```diff
@@ -33,38 +33,48 @@ exports[`the SearchBar component with headingText matches the snapshot 1`] = `
...
       id="kb-search-input"
       name="search-input"
       onChange={[Function]}
+      placeholder="Search the knowledge base"
       spellCheck="false"
       type="text"
     />
+    <button
+      className="sc-bwzfXH eeZBlJ sc-bdVaJa bsIiYo"
+      onClick={undefined}
```
Comment (IreneStr) [inline composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap]: undefined

File: composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap [hunk 12]
```diff
@@ -78,14 +88,24 @@ exports[`the SearchBar component without headingText matches the snapshot 1`] =
...
```
Comment (IreneStr) [inline composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap]: undefined

File: composites/AlgoliaSearch/SearchBar.js (hunk 1 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: OK then I'm going to update all of them, there are a few.

File: composites/AlgoliaSearch/SearchBar.js (hunk 4 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: I know it's not on the 8px grid, but this is the only value that matches the design in the provided screenshot.

File: composites/AlgoliaSearch/SearchBar.js (hunk 5 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: Same as above

File: composites/AlgoliaSearch/SearchBar.js (hunk 6 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: The proposed change doesn't tell exactly what I meant to warn about here. The same element is at the same time a flex item and a flex container. Will try to use the best of both worlds.

File: composites/AlgoliaSearch/SearchBar.js (hunk 7 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: I'm not sure we have standards for this :) this is not JavaScript, it's CSS. Anyways, I'll update all the occurrences across all files, there are a few...

File: composites/AlgoliaSearch/SearchBar.js (hunk 9 above)
Comment (afercia) [reply composites/AlgoliaSearch/SearchBar.js]: No 🙂

File: composites/AlgoliaSearch/tests/__snapshots__/AlgoliaSearcherTest.js.snap (hunk 10 above)
Comment (afercia) [reply composites/AlgoliaSearch/tests/__snapshots__/AlgoliaSearcherTest.js.snap]: Same as commented on #318, these are snapshots.

File: composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap (hunk 11 above)
Comment (afercia) [reply composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap]: Same as above

File: composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap (hunk 12 above)
Comment (afercia) [reply composites/AlgoliaSearch/tests/__snapshots__/SearchBarTest.js.snap]: Same as above

**Comment by IreneStr** [comment]
CR 👍

**Comment by IreneStr** [comment]
Acceptance 👍

**Pull request merged**